File: 409_longest_palindrome/409_longest_palindrome.go
package longestpalindrome

func longestPalindrome(s string) int {
	setOfCh := map[rune]struct{}{}
	cnt := 0

	for _, ch := range s {
		if _, present := setOfCh[ch]; !present {
			setOfCh[ch] = struct{}{}
		} else {
			delete(setOfCh, ch)
			cnt++
		}
	}
	if len(setOfCh) > 0 {
		return cnt*2 + 1
	}
	return cnt * 2

}

File: 409_longest_palindrome/409_longest_palindrome.ts
function longestPalindrome(s: string): number {
  let setOfCh: Set<string> = new Set<string>();
  let cnt: number = 0;

  for (let ch of s) {
    if (!setOfCh.has(ch)) {
      setOfCh.add(ch);
    } else {
      setOfCh.delete(ch);
      cnt++;
    }
  }
  if (setOfCh.size !== 0) {
    return cnt * 2 + 1;
  }
  return cnt * 2;
}
